File: projects/01/clojure/src/rectangle_area.clj
;;;; rectangle_area.clj calculates the area of a rectangle.
;;;;
;;;; Input: The width and height of a rectangle.
;;;; Output: The area of that rectangle.
;;;;
;;;; Usage: clojure -m rectangle_area
;;;;
;;;; Begun by: Prof. Adams, for CS 214 at Calvin College.
;;;; Completed by: Jacob Brink
;;;; Date: 2/12/2020
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns rectangle_area) ; namespace function names the program

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; Function rectangleArea() computes the area of a rectangle.
;;; Receive: itsWidth, itsHeight, a number.
;;; Precondition: itsHeight itsWidth  >= 0.0.
;;; Return: the area of the corresponding rectangle.
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
(defn rectangleArea [itsHeight itsWidth]
  (* itsHeight itsWidth 1.0 )  ; return width * height * 1.0 since formatting (%f) requires float not integer to display properly
)

(defn assertPositive [itsNumber]
  (assert (>= itsNumber 0) "-assertPositive: number must be positive")
  )

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; Function -main is where execution begins
;;; Input: the width and height of a rectangle.
;;; Output: the area of that rectangle.
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
(defn -main []
  (println "\nTo compute the area of a rectangle,")
  (print   " enter its height: ") (flush)
  (let
      [ height (read)]
      (assertPositive height)
      (print " enter its width: ") (flush)

      (let
        [ width (read) ]
        (assertPositive width)
        (printf "\nThe area is %f\n\n" (rectangleArea width height))
    )
    
  )
)  

File: projects/01/RectangleArea.java
/* RectangleArea.java calculates the area of a Rectangle
 *
 * Input: The width and height of a rectangle
 * Precondition: Width and height are positive
 * Output: The area of the rectangle
 *
 * Begun by: Dr. Adams, for CS 214, at Calvin College.
 * Completed by: Jacob Brink
 * Date: 2/6/2020
 **********************************************************/


import java.util.Scanner;  // include class for Scanner

public class RectangleArea {
    
     /* function rectangleArea() computes a rectangle's area, given its width and height.
      * Parameter: width and height, doubles
      * Precondition: width and height are positive
      * Returns: the area of the rectangle
      */
    public static double rectangleArea(double width, double height) {
        return width * height;               // return an expression
     } // rectangleArea method
	
  // main program
  public static void main(String[] args) {
      // prompt for width and height
      System.out.println("\n\nTo compute the area of a rectangle,");

      // Create a connection named keyboard to standard in
      Scanner keyboard = new Scanner(System.in);

      //Get width
      System.out.print(" enter its width: ");
      double width = keyboard.nextDouble();

      //Get height
      System.out.print(" enter its height: ");
      double height = keyboard.nextDouble();
      
      // output area
      System.out.println("\nThe area is " + rectangleArea(width, height) + "\n");
      System.out.printf("The area is %f\n\n", rectangleArea(width, height));
      System.out.printf("The area is %.15f\n\n", rectangleArea(width, height));
  } // main method

} // class RectangleArea

